sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamingJoinSuite.scala

```diff
@@ -425,6 +426,10 @@ class StreamingJoinSuite extends StreamTest with StateStoreMetricsTest with Befo
 
     // Test static comparisons
     assert(watermarkFrom("cast(leftTime AS LONG) > 10") === Some(10000))
+
+    // Test non-positive results
+    assert(watermarkFrom("CAST(leftTime AS LONG) > CAST(rightTime AS LONG) - 10") === Some(0))
+    assert(watermarkFrom("CAST(leftTime AS LONG) > CAST(rightTime AS LONG) - 100") === Some(-90000))
```

Comment:
These tests should be moved to a catalyst test suite (SteamingJoinHelperSuite) as the code it tests has moved to the StreamingJoinHelper.

Reply:
As mentioned earlier, I'm not sure how to move them; this test method as written relies on building a physical plan.

Reply:
It does not build a physical plan, it simply uses the SparkSQL QueryExecution to pass the expression through the analyzer and optimizer, to get the conditions in a similar form as a real query would. Instead we can simply using SimpleAnalyzer and SimpleTestOptimizer
```
    val analyzedPlan = SimpleAnalyzer.execute(plan)
    val optimizedPlan = SimpleTestOptimizer.execute(analyzedPlan)
```